provider/cluster/service.go

```diff
@@ -72,21 +76,39 @@ func NewService(ctx context.Context, session session.Session, bus pubsub.Bus, cl
 		return nil, err
 	}
 
-	hostnames := newHostnameService(ctx, cfg)
+	allHostnames, err := client.AllHostnames(ctx)
+	if err != nil {
+		sub.Close()
+		return nil, err
+	}
+
+	// Note: one side effect of this code is to add reservations for auto generated hostnames
+	// This is not normally done, but also doesn't cause any problems
```

Comment:
can they migrate those names?

Reply:
From what I remember if you try and deploy and SDL with a hostname in the autogenerated TLD it gets rejected entirely. I'll double check. It shouldn't be allowed.